src/include/sof/probe/probe.h

```diff
@@ -0,0 +1,80 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __SOF_PROBE_PROBE_H__
+#define __SOF_PROBE_PROBE_H__
+
+#include <ipc/probe.h>
+
+/*
+ * \brief Initialize probes subsystem
+ *
+ * param[in,optional] extraction_probe_dma - DMA associated with extraction
+ *		      In case extraction_probe_dma is NULL, extraction probes
+ *		      are unavailable.
+ */
+int probe_init(struct probe_dma *extraction_probe_dma);
+
+/*
+ * \brief Deinitialize probes subsystem.
+ *
+ * Detach extraction DMA if was enabled. Return -EINVAL in case some probes
+ * are still in use.
+ */
+int probe_deinit(void);
```

Comment:
same, what is the effect on injection DMAs?

Reply:
In current state (@akloniex correct me if I'm wrong) FW expects host side to perform cleanup phase before ending probe's life.

<pseudo>
array, probe_count = probe_point get()
while point_count
    if array[point_count] stream_tag matches this stream
        probe_point_disconnect(array[point_count])
    point_count--

if INJECTION
    array, dma_count = probe_dma get()
    while dma_count
         if array[dma_count] matches this stream
             probe_dma_detach(array[dma_count])
         dma_count--
</pseudo>

Reply:
@plbossart @crojewsk Correct. In case there are injection dmas, or any probe points still connected/active, deinit fails with -22.